pagefaultgames/pokerogue
# [Bug] Fix reloads erasing weather on first wave of biome

## What are the changes the user will see?

Reloading on the first wave of a new biome should no longer cause weather to disappear.

## Why am I making these changes?



Fixes #3877. This is also relevant to Daily Run Guides, which currently reload on the first wave of every new biome. 

## What are the changes from a developer perspective?


When entering a new biome, the game now generates the biome's weather before saving the session, which ensures that the weather gets saved correctly.
Also, `reloadHelper.ts` was updated to correctly use session data from the exact time session data gets saved, instead of the time when `reloadHelper` gets called.

## Screenshots/Videos



### Before the changes

https://github.com/user-attachments/assets/376bedb0-9011-4f2c-99cd-376aae02c8c7

After reloading, the weather is no longer Sunny, and Paras does not take damage from Dry Skin.

### After the changes

https://github.com/user-attachments/assets/b4a2b45d-ac3c-4868-89b7-4f4a73520109

After reloading, the weather remains Sunny, and Paras takes damage from Dry Skin.

## How to test the changes?




1. Enter a new biome that has weather. Take note of the type of weather.
2. Reload the session on the first wave of that biome.
3. Check that the weather is the same as before the reload.

## Checklist
